mono/metadata/gc.c

```diff
@@ -622,70 +629,73 @@ ves_icall_System_GC_WaitForPendingFinalizers (void)
 }
 
 void
-ves_icall_System_GC_register_ephemeron_array (MonoObject *array)
+ves_icall_System_GC_register_ephemeron_array (MonoObjectHandle array, MonoError *error)
 {
-#ifdef HAVE_SGEN_GC
-	if (!mono_gc_ephemeron_array_add (array)) {
-		mono_set_pending_exception (mono_object_domain (array)->out_of_memory_ex);
-		return;
-	}
-#endif
+	if (!mono_gc_ephemeron_array_add (MONO_HANDLE_RAW (array)))
+		mono_set_pending_exception (MONO_HANDLE_DOMAIN (array)->out_of_memory_ex);
```

Comment:
Could drop the use of `mono_set_pending_exception` in favor of `mono_error_set_exception_instance (error, ...); return;`

Reply:
Good point, I'll try.